reactive/beats_base.py

```diff
@@ -1,88 +1,77 @@
 from charms.layer import status
-from charms.reactive import when
-from charms.reactive import when_not
-from charms.reactive import set_state
 from charmhelpers.core.unitdata import kv
 
+from charms.reactive import (
+    when,
+    when_not,
+    set_state,
+)
 
-@when('config.changed')
+
+@when("config.changed")
 def config_changed():
-    set_state('beat.render')
+    set_state("beat.render")
 
 
-@when_not('logstash.available',
-          'elasticsearch.available',
-          'kafka.ready',
-          'config.set.logstash_hosts',
-          'config.set.kafka_hosts')
+@when_not("logstash.available",
+          "elasticsearch.available",
+          "kafka.ready",
+          "config.set.logstash_hosts",
+          "config.set.kafka_hosts")
 def waiting_messaging():
-    status.waiting('Waiting for: elasticsearch, logstash or kafka.')
+    status.waiting("Waiting for: elasticsearch, logstash or kafka.")
 
 
-@when('logstash.available')
-def cache_logstash_data(logstash):
-    units = logstash.list_unit_data()
-    cache = kv()
-    if cache.get('beat.logstash'):
-        hosts = cache.get('beat.logstash')
+def cache_data(host_name, host_instance):
```

Comment:
Can you call this service type rather than host_name

```suggestion
def cache_data(service, host_instance):
```